platform/darwin/src/MGLTileSet.h

```diff
@@ -0,0 +1,78 @@
+#import <Foundation/Foundation.h>
+#import "MGLTypes.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** These constants represent the scheme that the tile URL templates will use. */
+typedef NS_ENUM(NSUInteger, MGLTileSetScheme) {
+    MGLTileSetSchemeXYZ = 0,
+    MGLTileSetSchemeTMS
+};
+
+/**
+ The `MGLTileSet` class holds the tile URL template strings and associated
+ configuration for those strings. It can be passed to a `MGLVectorSource` or
+ `MGLRasterSource` instead of an `NSURL` representing a TileJSON URL to create a 
+ source.
+ */
+@interface MGLTileSet : NSObject
+
+/**
+ A REQUIRED `NSArray` of `NSString` objects that represent the tile templates.
```

Comment:
Optionality should be indicated through nullability qualifiers; shouting isn't needed here.